cpu/esp8266/periph/pwm.c

```diff
@@ -203,8 +203,12 @@ void pwm_set(pwm_t pwm, uint8_t channel, uint16_t value)
 
     next_off = next_on + value;
 
-    next_on = (_pwm_dev.cycles < next_on) ? next_on : next_on + _pwm_dev.res;
-    next_off = (_pwm_dev.cycles < next_off) ? next_off : next_off + _pwm_dev.res;
+    if (_pwm_dev.cycles < next_on) {
```

Comment:
Maybe this should be `if (_pwm_dev.cycles >= next_on) {`

Reply:
:confused: Of course, sorry changed it between two other tasks. Fixed.